src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12443,7 +12443,7 @@ void func_8084ECA4(Player* this, PlayState* play) {
     }
 
     if (this->skelAnime.curFrame <= 7.0f) {
-        this->stateFlags1 |= PLAYER_STATE1_1;
+        this->stateFlags1 |= PLAYER_STATE1_SWINGING_BOTTLE;
     }
 }
```

Comment:
somewhere in this function would be a good place to mark this bug https://www.youtube.com/watch?v=xTWSt7GJoSo
understand if you dont want to be the one to do that tho

Reply:
So the point here is that it `func_8083C0E8` will change the animation, so this can pass precisely once the final time this function runs for a swing?

Reply:
yea, there should have either been an early return after setting up the next action, or this block of setting the flag should have been higher up. i think both are sufficient fixes

Reply:
Yeah, I just wanted to make sure I understood the exact nature the glitch, I saw what you said about fixes in the video.